File: Spark/src/main/java/lxpsee/top/spark/core/study/scalaVerson/study4core/LineCountLocal.scala
package lxpsee.top.spark.core.study.scalaVerson.study4core

import lxpsee.top.spark.core.study.constant.LPConstants
import org.apache.spark.{SparkConf, SparkContext}

/**
  * The world always makes way for the dreamer
  * Created by 努力常态化 on 2019/3/27 10:39.
  */
object LineCountLocal {
  def main(args: Array[String]): Unit = {
    val context = new SparkContext(new SparkConf().setMaster("local").setAppName("LineCountLocal"))
    val lines = context.textFile(LPConstants.HELLO_TXT_LOCAL_FILE_PATH)
    val lineCount = lines.map((_, 1)).reduceByKey(_ + _)
    lineCount.foreach(t => println(t._1 + " 出现 " + t._2 + "次"))
  }
}

File: Spark/src/main/java/lxpsee/top/spark/core/study/javaVerson/study4core/LineCountLocal.java
package lxpsee.top.spark.core.study.javaVerson.study4core;

import lxpsee.top.spark.core.study.constant.LPConstants;
import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaPairRDD;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.Function2;
import org.apache.spark.api.java.function.PairFunction;
import org.apache.spark.api.java.function.VoidFunction;
import scala.Tuple2;

/**
 * The world always makes way for the dreamer
 * Created by 努力常态化 on 2019/3/27 10:34.
 */
public class LineCountLocal {
    public static void main(String[] args) {
        JavaSparkContext sparkContext = new JavaSparkContext(new SparkConf()
                .setMaster("local").setAppName("LineCount"));
        JavaRDD<String> lines = sparkContext.textFile(LPConstants.HELLO_TXT_LOCAL_FILE_PATH);
        JavaPairRDD<String, Integer> lineRDD = lines.mapToPair(new PairFunction<String, String, Integer>() {
            public Tuple2<String, Integer> call(String s) throws Exception {
                return new Tuple2<String, Integer>(s, 1);
            }
        });
        JavaPairRDD<String, Integer> lineCount = lineRDD.reduceByKey(new Function2<Integer, Integer, Integer>() {
            public Integer call(Integer v1, Integer v2) throws Exception {
                return v1 + v2;
            }
        });
        lineCount.foreach(new VoidFunction<Tuple2<String, Integer>>() {
            public void call(Tuple2<String, Integer> tuple2) throws Exception {
                System.out.println(tuple2._1 + " 出现了 " + tuple2._2 + " 次");
            }
        });

        sparkContext.close();
    }
}
